__tests__/core/browser-service.test.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2020-present, Elastic NV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import { BrowserService } from '../../src/core/browser-service';
+import { chromium } from 'playwright-chromium';
+
+describe('BrowserService', () => {
+  let browserService: BrowserService;
+  beforeEach(() => {
+    browserService = new BrowserService({ port: 9323 });
+    browserService.init();
+  });
+
+  afterEach(async () => {
+    await browserService.dispose();
+  });
+
+  it('should create browser pages', async () => {
+    const browser = await chromium.connect({
+      wsEndpoint: 'ws://localhost:9323',
```

Comment:
can we add a field to the browserservice class to extract the endpoint, would be good to have it there and use it everywhere for tests.

Reply:
The browser service does not necessarily know on which interface it is accessible, e.g. if it's running inside docker, so I don't think having the wsEndpoint there is a good idea.

Reply:
Probably i wasn't clear. I meant this
```
export class BrowserService {
  closeCallbacks = [];
  proxyServer = createServer();
  options: Record<string, unknown>;
  host: string;

  constructor(options = { port: 9322 }) {
    this.options = options;
    this.host = `ws:localhost:${this.options.port}`;
  }
}

```

We could use the `host` instead everywhere.

Reply:
Yeah, that's what I meant as well, the browser service isn't necessarily hosted on `localhost` therefore having the host on this class is not appropriate IMO.